overlord/hookstate/hookmgr.go

```diff
@@ -277,9 +288,21 @@ func snapCmd() string {
 	return filepath.Join(filepath.Dir(exe), "../../bin/snap")
 }
 
-func runHookAndWait(snapName string, revision snap.Revision, hookName, hookContext string, tomb *tomb.Tomb) ([]byte, error) {
+func killemAll(cmd *exec.Cmd) error {
+	pgid, err := syscall.Getpgid(cmd.Process.Pid)
+	if err != nil {
+		return err
+	}
+	return syscall.Kill(-pgid, 9)
+}
+
+func runHookAndWait(snapName string, revision snap.Revision, hookName, hookContext string, maxRuntime time.Duration, tomb *tomb.Tomb) ([]byte, error) {
 	command := exec.Command(snapCmd(), "run", "--hook", hookName, "-r", revision.String(), snapName)
 
+	// setup a process group for the command so that we can kill parent
```

Comment:
I was thinking about systemd-run as well when writing this. I'm not sure though if the extra work is needed. The upside is that it looks very nice in systemctl and is reliable to kill.